ignite/engine/engine.py

```diff
@@ -807,6 +807,7 @@ def _run_once_on_dataset(self) -> float:
                 )
 
             while True:
+                self.state.batch = self.state.output = None
```

Comment:
Awesome! I was thinking about that and wanted to comment out about that and you just implemented it 👍 
Thanks !